locator/tablets.cc

```diff
@@ -284,6 +284,15 @@ const tablet_map& tablet_metadata::get_tablet_map(table_id id) const {
     }
 }
 
+future<tablet_metadata::tablet_map_ptr> tablet_metadata::get_tablet_map_ptr(table_id id) const {
+    try {
+        // lightweight since it only copies the shared ptr, not the map itself.
+        co_return co_await _tablets.at(id).copy();
+    } catch (const std::out_of_range&) {
+        throw_with_backtrace<no_such_tablet_map>(id);
```

Comment:
Shouldn't we cancel the split compaction if tablet split is cancelled?

Reply:
> Shouldn't we cancel the split compaction if tablet split is cancelled?

I don't think cancelling split is a guarantee because it happens in background, and a merge can still happen while the job is being cancelled. So I decided to make split stable first, fixing the problem observed, and later we implement split cancelling properly, which will cancel all split jobs in background since the reaction to token metadata changes happens in a non futurized path. At the end of the day, we'd have both fixes, I implemented this one first.